server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -641,6 +641,11 @@ protected QueryDataSet processDataQuery(QueryPlan queryPlan, QueryContext contex
     queryDataSet.setRowOffset(queryPlan.getRowOffset());
     queryDataSet.setWithoutAllNull(queryPlan.isWithoutAllNull());
     queryDataSet.setWithoutAnyNull(queryPlan.isWithoutAnyNull());
+    // AlignByDevicePlan's withoutNullColumnsIndex has been set in the method
+    // `getAlignByDeviceDataSet` above
+    if (!(queryPlan instanceof AlignByDevicePlan)) {
+      queryDataSet.setWithoutNullColumnsIndex(queryPlan.getWithoutNullColumnsIndex());
+    }
```

Comment:
Actually, it's not necessary to use `if else` here. Because the `withoutNullColumnsIndex` should NOT be calculated in AlignByDeviceDataset, but in `generateAlignByDevicePlan` in QueryOperator.

Reply:
fix